### PR TITLE
Enable typing for external users

### PR DESCRIPTION
**Pull request by GrozniyToaster**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes

Enable typing for external users by add `py.typed` marker. 

[Enable typing docs](https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports)

### CODE REVIEW

File: etna/libs/tsfresh/relevance.pyi [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from typing import Optional, List
```
Comment (martins0n) [inline etna/libs/tsfresh/relevance.pyi:1]: We import every object seperately

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,6 +35,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - 
 - 
 ### Fixed
+- Type hints for external users by [PEP 561](https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-library-stubs-or-py-typed-marker)
```
Comment (martins0n) [inline CHANGELOG.md:38]: Can you add reference to PR as in other places?

File: CHANGELOG.md [hunk 3]
```diff
@@ -35,6 +35,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+-
```
Comment (martins0n) [inline CHANGELOG.md:39]: You shouldn't add new placeholders. It would cause merge conflicts

**Review by martins0n — CHANGES_REQUESTED** [review]

File: etna/libs/tsfresh/relevance.pyi (hunk 1 above)
Comment (GrozniyToaster) [reply etna/libs/tsfresh/relevance.pyi]: Done

File: CHANGELOG.md (hunk 2 above)
Comment (GrozniyToaster) [reply CHANGELOG.md]: done

**Review by martins0n — APPROVED** [review]

**Pull request merged**